content/en/blog/news/2022-gsoc-work-report.md

```diff
@@ -0,0 +1,127 @@
+  ```
+  ui:
+  proto: http
+  addr: 9200
+  schemes:
+    - http
+  ```
+- Add that domain to the ingress file, Run the following command and replace **dashboard.jx.change.me** with ngrok link.
+  ```
+  kubectl edit ing jx-ui
+  ```
+- And boom, visit localhost:9200 and it works, you should see the homepage screen.
+
+## Work Done
+### Stop a running or pending PipelineActivity from UI
+
+We have added a button in pipelines page and pipelineDetails page, it asks for confirmation and on selecting **Yes** it will stop the PipelineActivity.
+
+We can stop the PipelineActivity from the pipelines tables.
+![pipelinesPageStopBtn.png](/images/gsoc-work-report-22/pipelinesPageStopBtn.png)
+
+We can also stop the PipelineActivity from the pipelines details page.
+![pipelineDetailsPageStopBtn.png](/images/gsoc-work-report-22/pipelineDetailsPageStopBtn.png)
+
+**Issue:** https://github.com/jenkins-x/jx-ui/issues/28
+
+**PRs:**
+- https://github.com/jenkins-x/jx-ui/pull/50
+- https://github.com/jenkins-x/jx-ui/pull/51
+- https://github.com/jenkins-x/jx-ui/pull/59
+- https://github.com/jenkins-x/jx-ui/pull/75
+
+
+### Show message with status of the PipelineActivity
+
+When a pipeline is cancelled or it has timedout, we update the status of the pipeline to cancelled or timedout. But we dont show any reason why a job was timedout. We should show that information in the pipeline details page (along with the step which has timed out).
+
+**Issue:** https://github.com/jenkins-x/jx-ui/issues/23
+
+**PRs:**
+- https://github.com/jenkins-x/jx-api/pull/171
+- https://github.com/jenkins-x-plugins/jx-pipeline/pull/481
+
+
+### Implement a DAG for PipelineActivity
+
+To make the UI more user friendly and to explain the sequence of the steps in a PipelineActivity, we have added a DAG (Directed Acyclic Graph) chart to graphically visualize the PipelineActivity.
+
+We used [D3.js](https://d3js.org/) library to create a custom component which create a DAG chart of PipelineActivity for us. [D3-hierarchy](https://github.com/d3/d3-hierarchy) is an important part of our component. We're unit testing the component using [vitest](https://vitest.dev/).
+
+![pipelineDetailsPage.png](/images/gsoc-work-report-22/pipelineDetailsPage.png)
+
+**Issue:** https://github.com/jenkins-x/jx-ui/issues/48
+
+**PRs:**
+- https://github.com/jenkins-x/jx-ui/pull/57/files
+- https://github.com/jenkins-x/jx-ui/pull/65/files
+
+## What's next?
+
+- The UI project is just started, we have also start a seperate **UI-sig** which you can join from [calendar](https://jenkins-x.io/community/#office-hours).
+- There are some amazing features that we are planning to add to the new UI: 
+  - Dynamically update the list of pipelines
+  - Upgrade version stream from the UI
+  - Import repositories using the UI
+  - Break down pipeline logs into individual steps
+  - Mask secrets in the logs
+  - Store logs for cancelled and timed out pipelines
```

Comment:
Not directly related to the UI, as that needs to be implemented in the build controller ( or jx-pipeline codebase). So may be change this to `display logs for cancelled and timedout pipelines`